src/java/com/google/cloud/bigquery/dwhassessment/hooks/logger/utils/TasksRetriever.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022-2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigquery.dwhassessment.hooks.logger.utils;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.hadoop.hive.ql.exec.DDLTask;
+import org.apache.hadoop.hive.ql.exec.Task;
+
+/** Helper for retrieving tasks of the target type from the query plan tasks. */
+public final class TasksRetriever {
+  private TasksRetriever() {}
+
+  public static List<DDLTask> getDdlTasks(List<Task<? extends Serializable>> tasks) {
+    List<DDLTask> ddlTasks = new ArrayList<>();
+    if (tasks != null) {
+      getDdlTasksRecursively(tasks, ddlTasks);
+    }
+
+    return ddlTasks;
```

Comment:
The `ddlTasks` list will be iterated over multiple times, if there are dependent tasks for multiple tasks. What is the estimated max size of this list and the estimated number of tasks with dependent tasks?

Reply:
I initially copied the method body from the `Utilities.getMrTasks`, but since we don't really need to have an actual list of tasks here, I changed it to `hasDdlTask`.

Answering the exact question – based on my experiments, DDL queries always have 1 DDLTask without dependencies, but I wouldn't rely on that.